src/plots/plot.ts

```diff
@@ -451,9 +451,13 @@ export class Plot extends Component {
           return;
         }
 
+        const plot = this;
         lightweightPlotEntities.push({
           datum,
-          position,
+          get position() {
+            // only calculate position when needing to improve pan zoom performance #3159
+            return plot._pixelPoint.call(plot, datum, datumIndex, dataset);
```

Comment:
Cool work around for perf. This will likely not save us a huge amount since many methods up-convert "lightweight" entities to plot entities and i think that copies position over.

Still, it'll work great for interactions that compute with lightweight entities

Does this affect the serializability of lightweight plot entities vis a vis JSON.stringify?